source/Mlos.Python/mlos/Optimizers/RegressionModels/LassoCrossValidatedRegressionModel.py

```diff
@@ -0,0 +1,269 @@
+        self.regressor_standard_error_ = 0
+
+    @property
+    def trained(self):
+        return self._trained
+
+    @property
+    def num_observations_used_to_fit(self):
+        return self.last_refit_iteration_number
+
+    @property
+    def coef_(self):
+        if self.trained:
+            return self._regressor.coef_
+        return None
+
+    @property
+    def intercept_(self):
+        if self.trained:
+            return self._regressor.intercept_
+        return None
+
+    # TODO : the condition number of the p hat matrix is likely the ultimate factor in knowing if fit should be done
+    def should_fit(self, num_samples):
+        """ Returns true if the model should be fitted.
+
+        This model should be fitted under the following conditions:
+        1) It has not been fitted yet and num_samples is larger than min_samples_to_fit
+        2) The model has been fitted and the number of new samples is larger than n_new_samples_before_refit
+
+        :param num_samples:
+        :return:
+        """
+        num_input_dims = len(self.input_dimension_names)
+        model_config = self.model_config
+        if not self.trained:
+            return num_samples > model_config.min_num_samples_per_input_dimension_to_fit * num_input_dims
+        num_new_samples = num_samples - self.num_observations_used_to_fit
+        return num_new_samples >= model_config.num_new_samples_per_input_dimension_before_refit * num_input_dims
+
+    def _transform_x(self, x_df: DataFrame):
+        # confirm feature_values_pandas_frame contains all expected columns
+        #  if any are missing, impute NaN values
+        missing_column_names = set.difference(set(self.input_dimension_names), set(x_df.columns.values))
+        for missing_column_name in missing_column_names:
+            x_df[missing_column_name] = np.NaN
+
+        # impute 0s for NaNs (NaNs can come from hierarchical hypergrids)
+        x_df.fillna(value=0, inplace=True)
+
+        # construct traditional design matrix when fitting with one hot encoded categorical dimensions
+        if len(self.one_hot_encoder_adapter.get_one_hot_encoded_column_names()) > 0:
+            design_matrix = self._create_one_hot_encoded_design_matrix(x_df)
+        else:
+            design_matrix = x_df.to_numpy()
+        return design_matrix
+
+    @trace()
+    def fit(self, feature_values_pandas_frame, target_values_pandas_frame, iteration_number):
+        self.logger.debug(f"Fitting a {self.__class__.__name__} with {len(feature_values_pandas_frame.index)} observations.")
+
+        # Let's get the numpy arrays out of the panda frames
+        x_df = self.one_hot_encoder_adapter.project_dataframe(feature_values_pandas_frame, in_place=False)
+        y = target_values_pandas_frame[self.target_dimension_names].to_numpy()
+
+        design_matrix = self._transform_x(x_df)
+        # if len(self.one_hot_encoder_adapter.get_one_hot_encoded_column_names()) > 0:

```

Comment:
Is it OK to remove this?

Reply:
Yes.  Sorry I missed those commented out lines.  Pushed a cleaned up version.